packages/core-data/src/private-actions.js

```diff
@@ -14,3 +24,114 @@ export function receiveRegisteredPostMeta( postType, registeredPostMeta ) {
 		registeredPostMeta,
 	};
 }
+
+/**
+ * @typedef {Object} Modifier
+ * @property {string} [type] - The type of modifier.
+ * @property {Object} [args] - The arguments of the modifier.
+ */
+
+/**
+ * @typedef {Object} Edits
+ * @property {string}     [src]       - The URL of the media item.
+ * @property {Modifier[]} [modifiers] - The modifiers to apply to the media item.
+ */
+
+/**
+ * Duplicates a attachment / media entity record and, optionally, modifies it.
+ *
+ * @param {string}   kind                    Entity kind.
+ * @param {string}   name                    Entity name.
+ * @param {string}   recordId                Entity record ID.
+ * @param {Edits}    edits                   Edits to apply to the record.
+ * @param {Object}   options                 Options object.
+ * @param {Function} options.__unstableFetch Custom fetch function.
+ * @param {boolean}  options.throwOnError    Whether to throw an error if the request fails.
+ *
+ * @return {Promise} Promise resolving to the updated record.
+ */
+export const editMediaEntity =
+	(
+		kind,
+		name,
+		recordId,
+		edits = {},
+		{ __unstableFetch = apiFetch, throwOnError = false } = {}
+	) =>
+	async ( { dispatch, resolveSelect } ) => {
+		const configs = await resolveSelect.getEntitiesConfig( kind );
+		const entityConfig = configs.find(
+			( config ) => config.kind === kind && config.name === name
+		);
+
+		if ( ! entityConfig && entityConfig.baseURL !== '/wp/v2/media' ) {
+			return;
+		}
+
+		const path = `${ entityConfig.baseURL }/${ recordId }/edit`;
+
+		// Entity does not support duplication.
+		if ( ! path ) {
+			return;
+		}
```

Comment:
This is just a drive-by comment while I catch up on this PR, but it looks like this `if` condition will never be reached, because the `path` string will always be truthy. What's the desired check here?

Reply:
Definitely not that! 😄 It's an oversight.